browser/misc_metrics/doh_metrics.cc

```diff
@@ -136,14 +155,13 @@ void DohMetrics::OnDnsRequestCounts(
   double percentage =
       static_cast<double>(upgraded_request_storage_->GetWeeklySum()) /
       std::max(total_request_storage_->GetWeeklySum(), uint64_t(1u)) * 100.0;
-  if (percentage == 0.0) {
-    UMA_HISTOGRAM_EXACT_LINEAR(histogram_name, INT_MAX - 1, 4);
-    return;
+  int percentage_int = 0;
+  if (percentage > 0.0) {
```

Comment:
Is this check necessary? Or can you just use
`int percentage_int = static_cast<int>(percentage);`

Reply:
yes. i.e. casting 0.5% into an int would render 0, which would fall into the 0% bucket, when it should be in the "> 0% and <= 5%" bucket

Reply:
as discussed over DM, this was an error. `std::max(static_cast<int>(percentage), 1)` should have been used, or even `static_cast<int>(std::ceil(percentage))`. will leave as is for now.